plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -607,14 +589,15 @@ public ColumnMetadata getColumnMetadata(ConnectorSession session, ConnectorTable
     {
         DeltaLakeTableHandle table = (DeltaLakeTableHandle) tableHandle;
         DeltaLakeColumnHandle column = (DeltaLakeColumnHandle) columnHandle;
-        if (column.getProjectionInfo().isPresent()) {
+        if (column.getProjectionInfo().isPresent() || column.getColumnType() == SYNTHESIZED) {
             return getColumnMetadata(column, null, true, null);
         }
```

Comment:
Do we need this additional logic? Shouldn't `DeltaTable` construct correctly and return nulls for such case?

Reply:
I didn't want to include synthesized columns in `DeltaTable` so that the class throws exceptions if a specified column doesn't exist in the object.